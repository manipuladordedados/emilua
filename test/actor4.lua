if _CONTEXT == 'main' then
    local ch = spawn_vm('.')

    -- The number of yield()s required is an implementation detail and may
    -- change among versions.
    --
    -- The point is: The worker must finish before the main actor calls
    -- `ch:send()`. That's what we're testing: can the sender detect the
    -- receiver is already gone?
    this_fiber.yield()

    print('A')
    ch:send('foobar')
else
    assert(_CONTEXT == 'worker')
    print('B')
end
