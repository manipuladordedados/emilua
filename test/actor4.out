B
A
Main fiber from VM 0x1 panicked: 'Channel closed'
stack traceback:
	[string "?"]: in function 'send'
	input.lua:13: in main chunk
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
